From ff35fdfcbbc79daa6ec7a8a21c963f0668c9c820 Mon Sep 17 00:00:00 2001
From: Christophe Marin <christophe@krop.fr>
Date: Tue, 7 Oct 2025 14:14:43 +0200
Subject: Revert "Modify headers installation for CMake builds"

/system/develop/headers is the correct location for include headers.

diff --git a/build_scripts/platforms/unix.py b/build_scripts/platforms/unix.py
index 59c63fd..c5710dd 100644
--- a/build_scripts/platforms/unix.py
+++ b/build_scripts/platforms/unix.py
@@ -108,11 +108,11 @@ def prepare_packages_posix(pyside_build, _vars, cross_build=False):
             "{st_build_dir}/{st_package_name}/scripts/shiboken_tool.py",
             force=False, _vars=_vars)
 
-    if config.is_internal_shiboken_module_build() or config.is_internal_pyside_build():
-        # <install>/{cmake_package_name}/include/* -> <package_for_wheels>/{st_package_name}/include
+    if config.is_internal_shiboken_generator_build() or config.is_internal_pyside_build():
+        # <install>/include/* -> <setup>/{st_package_name}/include
         copydir(
             "{install_dir}/{cmake_package_name}/include",
-            destination_dir / "include",
+            destination_dir / "$includedir$",
             _vars=_vars)
 
     if config.is_internal_pyside_build():
diff --git a/sources/pyside6/PySide6/CMakeLists.txt b/sources/pyside6/PySide6/CMakeLists.txt
index c1934d9..4639cbe 100644
--- a/sources/pyside6/PySide6/CMakeLists.txt
+++ b/sources/pyside6/PySide6/CMakeLists.txt
@@ -109,17 +109,17 @@ install(FILES "${CMAKE_CURRENT_BINARY_DIR}/_git_pyside_version.py"
 install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/support"
         DESTINATION "${PYTHON_SITE_PACKAGES}/${BINDING_NAME}${pyside6_SUFFIX}")
 install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/templates/common.xml
-        DESTINATION share/PySide6${pyside_SUFFIX}/typesystems)
+        DESTINATION ${CMAKE_INSTALL_DATADIR}/PySide6${pyside_SUFFIX}/typesystems)
 install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/templates/core_common.xml
-        DESTINATION share/PySide6${pyside_SUFFIX}/typesystems)
+        DESTINATION ${CMAKE_INSTALL_DATADIR}/PySide6${pyside_SUFFIX}/typesystems)
 install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/templates/gui_common.xml
-        DESTINATION share/PySide6${pyside_SUFFIX}/typesystems)
+        DESTINATION ${CMAKE_INSTALL_DATADIR}/PySide6${pyside_SUFFIX}/typesystems)
 install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/templates/widgets_common.xml
-        DESTINATION share/PySide6${pyside_SUFFIX}/typesystems)
+        DESTINATION ${CMAKE_INSTALL_DATADIR}/PySide6${pyside_SUFFIX}/typesystems)
 install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/templates/datavisualization_common.xml
-        DESTINATION share/PySide6${pyside_SUFFIX}/typesystems)
+        DESTINATION ${CMAKE_INSTALL_DATADIR}/PySide6${pyside_SUFFIX}/typesystems)
 install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/pyside6_global.h
-        DESTINATION ${BINDING_NAME}${pyside6_SUFFIX}/include)
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${BINDING_NAME}${pyside6_SUFFIX})
 
 install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/QtAsyncio"
         DESTINATION "${PYTHON_SITE_PACKAGES}/${BINDING_NAME}${pyside6_SUFFIX}")
diff --git a/sources/pyside6/PySide6/QtCore/CMakeLists.txt b/sources/pyside6/PySide6/QtCore/CMakeLists.txt
index 2f49b61..39932cf 100644
--- a/sources/pyside6/PySide6/QtCore/CMakeLists.txt
+++ b/sources/pyside6/PySide6/QtCore/CMakeLists.txt
@@ -306,4 +306,4 @@ if (APPLE)
 endif()
 
 install(FILES ${pyside6_SOURCE_DIR}/qtcorehelper.h ${pyside6_SOURCE_DIR}/qiopipe.h
-        DESTINATION PySide6/include/QtCore/)
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/PySide6/QtCore/)
diff --git a/sources/pyside6/PySide6/QtDBus/CMakeLists.txt b/sources/pyside6/PySide6/QtDBus/CMakeLists.txt
index 49e86d1..10945ef 100644
--- a/sources/pyside6/PySide6/QtDBus/CMakeLists.txt
+++ b/sources/pyside6/PySide6/QtDBus/CMakeLists.txt
@@ -51,4 +51,4 @@ create_pyside_module(NAME QtDBus
                      TYPESYSTEM_PATH QtDBus_SOURCE_DIR
                      SOURCES QtDBus_SRC)
 
-install(FILES ${pyside6_SOURCE_DIR}/qtdbushelper.h DESTINATION PySide6/include/QtDBus)
+install(FILES ${pyside6_SOURCE_DIR}/qtdbushelper.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/PySide6/QtDBus)
diff --git a/sources/pyside6/PySide6/QtDataVisualization/CMakeLists.txt b/sources/pyside6/PySide6/QtDataVisualization/CMakeLists.txt
index 424799b..6f81976 100644
--- a/sources/pyside6/PySide6/QtDataVisualization/CMakeLists.txt
+++ b/sources/pyside6/PySide6/QtDataVisualization/CMakeLists.txt
@@ -68,4 +68,4 @@ create_pyside_module(NAME QtDataVisualization
                      STATIC_SOURCES QtDataVisualization_src)
 
 install(FILES ${pyside6_SOURCE_DIR}/qtdatavisualization_helper.h
-        DESTINATION PySide6/include/QtDataVisualization)
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/PySide6/QtDataVisualization)
diff --git a/sources/pyside6/PySide6/QtDesigner/CMakeLists.txt b/sources/pyside6/PySide6/QtDesigner/CMakeLists.txt
index 570ed0f..aba20d1 100644
--- a/sources/pyside6/PySide6/QtDesigner/CMakeLists.txt
+++ b/sources/pyside6/PySide6/QtDesigner/CMakeLists.txt
@@ -67,4 +67,4 @@ create_pyside_module(NAME QtDesigner
                      STATIC_SOURCES QtDesigner_static_src
                      TYPESYSTEM_NAME ${QtDesigner_BINARY_DIR}/typesystem_designer.xml)
 
-install(FILES ${pyside6_SOURCE_DIR}/qpydesignerextensions.h DESTINATION PySide6/include/QtDesigner)
+install(FILES ${pyside6_SOURCE_DIR}/qpydesignerextensions.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/PySide6/QtDesigner)
diff --git a/sources/pyside6/PySide6/QtGraphs/CMakeLists.txt b/sources/pyside6/PySide6/QtGraphs/CMakeLists.txt
index f416078..687f929 100644
--- a/sources/pyside6/PySide6/QtGraphs/CMakeLists.txt
+++ b/sources/pyside6/PySide6/QtGraphs/CMakeLists.txt
@@ -96,4 +96,4 @@ create_pyside_module(NAME QtGraphs
                      DROPPED_ENTRIES QtGraphs_DROPPED_ENTRIES)
 
 install(FILES ${pyside6_SOURCE_DIR}/qtgraphs_helper.h
-        DESTINATION PySide6/include/QtGraphs)
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/PySide6/QtGraphs)
diff --git a/sources/pyside6/PySide6/QtGui/CMakeLists.txt b/sources/pyside6/PySide6/QtGui/CMakeLists.txt
index b9e8429..0336558 100644
--- a/sources/pyside6/PySide6/QtGui/CMakeLists.txt
+++ b/sources/pyside6/PySide6/QtGui/CMakeLists.txt
@@ -366,4 +366,4 @@ create_pyside_module(NAME QtGui
 
 install(FILES ${pyside6_SOURCE_DIR}/qpytextobject.h
               ${pyside6_SOURCE_DIR}/qtguihelper.h
-        DESTINATION PySide6/include/QtGui/)
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/PySide6/QtGui/)
diff --git a/sources/pyside6/cmake/Macros/PySideModules.cmake b/sources/pyside6/cmake/Macros/PySideModules.cmake
index 86791f4..e5bf964 100644
--- a/sources/pyside6/cmake/Macros/PySideModules.cmake
+++ b/sources/pyside6/cmake/Macros/PySideModules.cmake
@@ -172,7 +172,7 @@ macro(create_pyside_module)
     set(${module_NAME}_glue "${CMAKE_CURRENT_SOURCE_DIR}/../glue/${lower_module_name}.cpp")
     set(${module_name}_glue_dependency "")
     if(EXISTS ${${module_NAME}_glue})
-        install(FILES ${${module_NAME}_glue} DESTINATION share/PySide6${pyside6_SUFFIX}/glue)
+        install(FILES ${${module_NAME}_glue} DESTINATION ${CMAKE_INSTALL_DATADIR}/PySide6${pyside6_SUFFIX}/glue)
         set(${module_NAME}_glue_dependency ${${module_NAME}_glue})
     endif()
 
@@ -180,14 +180,14 @@ macro(create_pyside_module)
     set(${module_NAME}_doc "${CMAKE_CURRENT_SOURCE_DIR}/../doc/${lower_module_name}.rst")
     set(${module_name}_doc_dependency "")
     if(EXISTS ${${module_NAME}_doc})
-        install(FILES ${${module_NAME}_doc} DESTINATION share/PySide6${pyside6_SUFFIX}/doc)
+        install(FILES ${${module_NAME}_doc} DESTINATION ${CMAKE_INSTALL_DATADIR}/PySide6${pyside6_SUFFIX}/doc)
         set(${module_NAME}_doc_dependency ${${module_NAME}_doc})
     endif()
 
     # Install standalone glue files into typesystems subfolder, so that the resolved relative
     # paths remain correct.
     if (module_GLUE_SOURCES)
-        install(FILES ${module_GLUE_SOURCES} DESTINATION share/PySide6${pyside6_SUFFIX}/typesystems/glue)
+        install(FILES ${module_GLUE_SOURCES} DESTINATION ${CMAKE_INSTALL_DATADIR}/PySide6${pyside6_SUFFIX}/typesystems/glue)
     endif()
 
     shiboken_get_tool_shell_wrapper(shiboken tool_wrapper)
@@ -378,7 +378,7 @@ macro(create_pyside_module)
 
 
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/PySide6/${module_NAME}/pyside6_${lower_module_name}_python.h
-            DESTINATION PySide6${pyside6_SUFFIX}/include/${module_NAME}/)
+            DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/PySide6${pyside6_SUFFIX}/${module_NAME}/)
     file(GLOB typesystem_files ${CMAKE_CURRENT_SOURCE_DIR}/typesystem_*.xml ${typesystem_path})
 
 #   Copy typesystem files and remove module names from the <load-typesystem> element
@@ -391,7 +391,7 @@ macro(create_pyside_module)
         string(REGEX REPLACE "<load-typesystem name=\"[^/\"]+/" "<load-typesystem name=\"" typesystemXml "${typesystemXml}")
         set (typesystem_target_file "${CMAKE_CURRENT_BINARY_DIR}/PySide6/typesystems/${typesystem_file_name}")
         file(WRITE "${typesystem_target_file}" "${typesystemXml}")
-        install(FILES "${typesystem_target_file}" DESTINATION share/PySide6${pyside6_SUFFIX}/typesystems)
+        install(FILES "${typesystem_target_file}" DESTINATION ${CMAKE_INSTALL_DATADIR}/PySide6${pyside6_SUFFIX}/typesystems)
     endforeach()
 endmacro()
 
diff --git a/sources/pyside6/libpyside/CMakeLists.txt b/sources/pyside6/libpyside/CMakeLists.txt
index b0050dd..15c88e0 100644
--- a/sources/pyside6/libpyside/CMakeLists.txt
+++ b/sources/pyside6/libpyside/CMakeLists.txt
@@ -101,7 +101,7 @@ append_size_optimization_flags(pyside6)
 
 target_include_directories(pyside6 PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-  $<INSTALL_INTERFACE:PySide6/include>
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/PySide6>
 )
 
 target_compile_definitions(pyside6 PRIVATE -DQT_LEAN_HEADERS=1 -DQT_NO_KEYWORDS=1)
@@ -147,8 +147,8 @@ include(CMakePackageConfigHelpers)
 
 # Build-tree / super project package config file.
 set(PYSIDE_PYTHONPATH "${PYTHON_SITE_PACKAGES}/PySide6")
-set(PYSIDE_TYPESYSTEMS "${CMAKE_INSTALL_PREFIX}/share/PySide6${pyside6_SUFFIX}/typesystems")
-set(PYSIDE_GLUE "${CMAKE_INSTALL_PREFIX}/share/PySide6${pyside6_SUFFIX}/glue")
+set(PYSIDE_TYPESYSTEMS "${CMAKE_INSTALL_DATADIR}/PySide6${pyside6_SUFFIX}/typesystems")
+set(PYSIDE_GLUE "${CMAKE_INSTALL_DATADIR}/PySide6${pyside6_SUFFIX}/glue")
 
 configure_package_config_file(
     "${CMAKE_CURRENT_SOURCE_DIR}/PySide6Config-spec.cmake.in"
@@ -177,7 +177,7 @@ configure_file("${CMAKE_CURRENT_SOURCE_DIR}/PySide6ConfigVersion.cmake.in"
                "${CMAKE_CURRENT_BINARY_DIR}/PySide6ConfigVersion.cmake" @ONLY)
 
 install(FILES ${libpyside_HEADERS}
-        DESTINATION ${BINDING_NAME}${pyside6_SUFFIX}/include)
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${BINDING_NAME}${pyside6_SUFFIX})
 
 # build-time installation
 install(TARGETS pyside6 EXPORT PySide6Targets
diff --git a/sources/pyside6/libpyside/pyside6.pc.in b/sources/pyside6/libpyside/pyside6.pc.in
index 65822e6..6c53ac4 100644
--- a/sources/pyside6/libpyside/pyside6.pc.in
+++ b/sources/pyside6/libpyside/pyside6.pc.in
@@ -1,9 +1,9 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=@CMAKE_INSTALL_PREFIX@
 libdir=@LIB_INSTALL_DIR@
-includedir=@CMAKE_INSTALL_PREFIX@/PySide6@pyside6_SUFFIX@/include
-typesystemdir=@CMAKE_INSTALL_PREFIX@/share/PySide6@pyside6_SUFFIX@/typesystems
-gluedir=@CMAKE_INSTALL_PREFIX@/share/PySide6@pyside6_SUFFIX@/glue
+includedir=@CMAKE_INSTALL_INCLUDEDIR@/PySide6@pyside6_SUFFIX@
+typesystemdir=@CMAKE_INSTALL_DATADIR@/PySide6@pyside6_SUFFIX@/typesystems
+gluedir=@CMAKE_INSTALL_INCLUDEDIR@/PySide6@pyside6_SUFFIX@/glue
 pythonpath=@PYTHON_SITE_PACKAGES@
 
 Name: PySide6@pyside6_SUFFIX@
diff --git a/sources/pyside6/libpysideqml/CMakeLists.txt b/sources/pyside6/libpysideqml/CMakeLists.txt
index 1430334..fe81d41 100644
--- a/sources/pyside6/libpysideqml/CMakeLists.txt
+++ b/sources/pyside6/libpysideqml/CMakeLists.txt
@@ -84,7 +84,7 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D QT_NO_CAST_FROM_ASCII -D QT_NO_CAST_T
 qfp_strip_library("pyside6qml")
 
 install(FILES ${libpysideqml_HEADERS}
-        DESTINATION ${BINDING_NAME}${pyside6qml_SUFFIX}/include)
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${BINDING_NAME}${pyside6qml_SUFFIX})
 
 #built-time installation
 install(TARGETS pyside6qml EXPORT PySide6Targets
diff --git a/sources/pyside6/libpysideremoteobjects/CMakeLists.txt b/sources/pyside6/libpysideremoteobjects/CMakeLists.txt
index caaadc2..a8f6e98 100644
--- a/sources/pyside6/libpysideremoteobjects/CMakeLists.txt
+++ b/sources/pyside6/libpysideremoteobjects/CMakeLists.txt
@@ -80,7 +80,7 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D QT_NO_CAST_FROM_ASCII -D QT_NO_CAST_T
 #
 
 install(FILES ${libpysideremoteobjects_HEADERS}
-        DESTINATION ${BINDING_NAME}${pyside6remoteobjects_SUFFIX}/include)
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${BINDING_NAME}${pyside6remoteobjects_SUFFIX})
 
 install(TARGETS pyside6remoteobjects EXPORT PySide6RemoteObjectsTargets
                            LIBRARY DESTINATION "${LIB_INSTALL_DIR}"
-- 
2.56.0


From c8e6aa6dc971b82f58a2ba831b8c06fcf1cf264e Mon Sep 17 00:00:00 2001
From: Luc Schrijvers <begasus@gmail.com>
Date: Wed, 16 Apr 2025 15:09:57 +0200
Subject: Fix for pthread_self()


diff --git a/sources/shiboken6/libshiboken/helper.cpp b/sources/shiboken6/libshiboken/helper.cpp
index 016ba12..7624f80 100644
--- a/sources/shiboken6/libshiboken/helper.cpp
+++ b/sources/shiboken6/libshiboken/helper.cpp
@@ -600,7 +600,7 @@ ThreadId currentThreadId()
 {
 #if defined(_WIN32)
     return GetCurrentThreadId();
-#elif defined(__APPLE_CC__)
+#elif defined(__APPLE_CC__) || defined(__HAIKU__)
     return reinterpret_cast<ThreadId>(pthread_self());
 #else
     return pthread_self();
-- 
2.56.0


From 1502a8571072bbf683e6f73897b0ab791bc37e6e Mon Sep 17 00:00:00 2001
From: Luc Schrijvers <begasus@gmail.com>
Date: Sat, 17 Oct 2026 18:02:35 +0200
Subject: fix include installation of shiboken6

This was part of the previous commit before.

diff --git a/sources/shiboken6/libshiboken/CMakeLists.txt b/sources/shiboken6/libshiboken/CMakeLists.txt
index 85f9433..f7a88b1 100644
--- a/sources/shiboken6/libshiboken/CMakeLists.txt
+++ b/sources/shiboken6/libshiboken/CMakeLists.txt
@@ -111,7 +111,7 @@ add_library(Shiboken6::libshiboken ALIAS libshiboken)
 target_include_directories(libshiboken PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
-  $<INSTALL_INTERFACE:shiboken6/include>
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/shiboken6>
 )
 
 if (NOT "${NUMPY_INCLUDE_DIR}" STREQUAL "")
@@ -197,7 +197,7 @@ install(FILES
         signature.h
         signature_p.h
 
-        DESTINATION shiboken6${shiboken6_SUFFIX}/include)
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/shiboken6${shiboken6_SUFFIX})
 install(TARGETS libshiboken EXPORT Shiboken6Targets
                             LIBRARY DESTINATION "${LIB_INSTALL_DIR}"
                             ARCHIVE DESTINATION "${LIB_INSTALL_DIR}"
-- 
2.56.0


From a650e232e0590c70ebaedb07532c6252549630d5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Joachim=20Mairb=C3=B6ck?= <j.mairboeck@gmail.com>
Date: Sat, 17 Oct 2026 17:57:23 +0200
Subject: fix finding Shiboken6 when building it together with pyside6 with
 cmake


diff --git a/sources/pyside6/cmake/Macros/PySideModules.cmake b/sources/pyside6/cmake/Macros/PySideModules.cmake
index e5bf964..fa5ada6 100644
--- a/sources/pyside6/cmake/Macros/PySideModules.cmake
+++ b/sources/pyside6/cmake/Macros/PySideModules.cmake
@@ -346,7 +346,11 @@ macro(create_pyside_module)
     # on the host machine (usually, unless you use some userspace qemu based mechanism).
     # TODO: Can we do something better here to still get pyi files?
     if(NOT (PYSIDE_IS_CROSS_BUILD OR DISABLE_PYI))
-        set(SHIBOKEN_PYTHON_MODULE_DIR "${PYTHON_SITE_PACKAGES}/shiboken6")
+        if(shiboken6_BINARY_DIR)
+            set(SHIBOKEN_PYTHON_MODULE_DIR "${shiboken6_BINARY_DIR}")
+        else()
+            set(SHIBOKEN_PYTHON_MODULE_DIR "${PYTHON_SITE_PACKAGES}/shiboken6")
+        endif()
         set(generate_pyi_options ${module_NAME} --sys-path
             "${pysidebindings_BINARY_DIR}"
             "${SHIBOKEN_PYTHON_MODULE_DIR}/.."
-- 
2.56.0


From 3af5b8f3e3a52008e30ff4b82a743d528a7343fd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Joachim=20Mairb=C3=B6ck?= <j.mairboeck@gmail.com>
Date: Sat, 17 Oct 2026 19:36:52 +0200
Subject: fix shebang for Haiku


diff --git a/sources/pyside-tools/pyside_tool.py b/sources/pyside-tools/pyside_tool.py
index 5c00985..15eb46a 100644
--- a/sources/pyside-tools/pyside_tool.py
+++ b/sources/pyside-tools/pyside_tool.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!python3
 # Copyright (C) 2022 The Qt Company Ltd.
 # SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
 from __future__ import annotations
diff --git a/sources/shiboken6/shiboken_tool.py b/sources/shiboken6/shiboken_tool.py
index e1b4b68..caabda8 100755
--- a/sources/shiboken6/shiboken_tool.py
+++ b/sources/shiboken6/shiboken_tool.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!python3
 # Copyright (C) 2022 The Qt Company Ltd.
 # SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
 from __future__ import annotations
-- 
2.56.0

